src/libraries/System.Threading.Tasks.Dataflow/src/PACKAGE.md

```diff
@@ -2,43 +2,143 @@
 
 <!-- A description of the package and where one can find more documentation -->
 
+Provides dataflow components that are collectively referred to as the *TPL Dataflow Library*.
+This dataflow model promotes actor-based programming by providing in-process message passing for coarse-grained dataflow and pipelining tasks.
 
+Note: This package is part of the larger *Task Parallel library* and is designed to help increase the robustness of concurrency-enabled applications.
 
 ## Key Features
 
 <!-- The key features of this package -->
 
-*
-*
-*
+* Foundation for message passing and parallelizing CPU-intensive and I/O-intensive applications that have high throughput and low latency.
+* Provides multiple block types for various dataflow operations (e.g., `BufferBlock`, `ActionBlock`, `TransformBlock`).
+* Blocks support linking to form *networks*, allowing for the creation of complex processing topologies.
 
 ## How to Use
 
 <!-- A compelling example on how to use this package with code, as well as any specific guidelines for when to use the package -->
 
+This sample demonstrates a dataflow pipeline that downloads the book "The Iliad of Homer" from a website and searches the text to match individual words with words that reverse the first word's characters.
+
+```csharp
+using System.Net;
+using System.Threading.Tasks.Dataflow;
+
+// Setup blocks
+
+// Downloads the requested resource as a string.
+TransformBlock<string, string> downloadString = new TransformBlock<string, string>(async uri =>
+{
+    Console.WriteLine("Downloading '{0}'...", uri);
+
+    return await new HttpClient(new HttpClientHandler { AutomaticDecompression = DecompressionMethods.GZip }).GetStringAsync(uri);
```

Comment:
We shouldn't have a sample that creates a new HttpClient{Handler} for every individual message.  The HttpClient should be cached into a static readonly field as a singleton used for every message.

Reply:
`static readonly` is not available in a non `Program.Main` style console app, but I put the creation of the handler outside the task.